src/ExecClient.cs

```diff
@@ -0,0 +1,116 @@
+using System;
+using System.Buffers;
+using System.Net.WebSockets;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace k8s
+{
+    public class ExecClient : IDisposable
+    {
+        private readonly ClientWebSocket socket;
+        private readonly CancellationTokenSource cts;
+        private Task runLoop;
+
+        public ExecClient(ClientWebSocket socket)
+        {
+            this.socket = socket ?? throw new ArgumentNullException(nameof(socket));
+            this.cts = new CancellationTokenSource();
+        }
+
+        public event EventHandler ConnectionClosed;
+        public event EventHandler<string> StandardOutputReceived;
+        public event EventHandler<string> StandardErrorReceived;
+
+        public void Start()
+        {
+            if (this.runLoop == null)
+            {
+                this.runLoop = this.RunLoop(this.cts.Token);
+            }
+        }
+
+        public void Stop(CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (this.runLoop != null)
+            {
+                this.cts.Cancel();
+                this.runLoop.Wait(cancellationToken);
+            }
+        }
+
+        public void Dispose()
+        {
+            this.Stop();
+        }
+
+        public async Task Write(string command, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            // Send:
+            //  - [uint32: stream index = 0 for stdin]
+            //  - [byte[]: command as UTF8 value]
+            var commandLength = Encoding.UTF8.GetByteCount(command);
+            var buffer = ArrayPool<byte>.Shared.Rent(commandLength + 4);
+
+            try
```

Comment:
nit: may as well move this try down to around the send, right?

Reply:
The try/finally protects `buffer` which comes from the shared array pool. The code that initializes the first 4 bytes of the buffer won't fail, but `Encoding.UTF8.GetBytes` may throw an exception. So if we move the `try` down to `SendAsync`, we may leak a buffer if `GetBytes` fails.
(It's very unlikely it will since we already call `Encoding.UTF8.GetByteCount`)

I can move it down; but because the `try` is there make sure we return `buffer` once we're done with it, I prefer to keep it close to `buffer`. 

I don't have a strong preference, though, so can always move it down if you prefer it that way.